cmd/cosign/cli/env_test.go

```diff
@@ -16,22 +16,204 @@
 package cli
 
 import (
+	"io"
 	"os"
 	"testing"
 
-	"github.com/google/go-cmp/cmp"
+	"github.com/sigstore/cosign/pkg/cosign/env"
 )
 
-func TestGetEnv(t *testing.T) {
-	os.Setenv("COSIGN_EXPERIMENTAL", "foo")
-	os.Setenv("TUF_ROOT", "bar")
-	got := getEnv()
-	want := []string{
-		"COSIGN_EXPERIMENTAL=foo",
-		"TUF_ROOT=bar",
+const (
+	VariableTest1 env.Variable = "COSIGN_TEST1"
+	VariableTest2 env.Variable = "COSIGN_TEST2"
+
+	expectedWithoutDescription = `COSIGN_TEST1="abcd"
+COSIGN_TEST2=""
+`
+	expectedWithDescription = `# COSIGN_TEST1 is the first test variable
+# Expects: test1 value
+COSIGN_TEST1="abcd"
+# COSIGN_TEST2 is the second test variable
+# Expects: test2 value
+COSIGN_TEST2=""
+`
+
+	expectedWithHiddenSensitive = `# COSIGN_TEST1 is the first test variable
+# Expects: test1 value
+COSIGN_TEST1="abcd"
+# COSIGN_TEST2 is the second test variable
+# Expects: test2 value
+COSIGN_TEST2="******"
+`
+
+	expectedWithSensitive = `# COSIGN_TEST1 is the first test variable
+# Expects: test1 value
+COSIGN_TEST1="abcd"
+# COSIGN_TEST2 is the second test variable
+# Expects: test2 value
+COSIGN_TEST2="1234"
+`
+
+	expectedSensitiveWithoutDescription = `COSIGN_TEST1="abcd"
+COSIGN_TEST2="1234"
+`
+
+	expectedWithNonRegisteredEnv = `# COSIGN_TEST1 is the first test variable
+# Expects: test1 value
+COSIGN_TEST1="abcd"
+# COSIGN_TEST2 is the second test variable
+# Expects: test2 value
+COSIGN_TEST2=""
+# Environment variables below are not registered with cosign,
+# but might still influence cosign's behavior.
+COSIGN_TEST3=abcd
```

Comment:
Hm...I think the safe thing to do is to treat everything as sensitive unless told otherwise (`--show-sensitive`).

I could just `export COSIGN_MY_ROOT_PASSWORD=hunter2` and it'd get printed out. That's *kind of* my fault, but 🤷. You could imagine a situation where `sigstore/sigstore` adds a new variable for a password and we don't put it into our map by the next Cosign release, which would be actually bad.

It's a little bit user-unfriendly but definitely safe.

Reply:
I agree it's definitely safer. It should be fixed now. I also added a few test cases for this.